src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -421,8 +407,16 @@ def ensure_container_insights_for_monitoring(
             if data_collection_settings is not None:
                 dataCollectionSettings = _get_data_collection_settings(data_collection_settings)
                 validate_data_collection_settings(dataCollectionSettings)
+                dataCollectionSettings.setdefault("enableContainerLogV2", True)
                 extensionSettings["dataCollectionSettings"] = dataCollectionSettings
                 cistreams = dataCollectionSettings["streams"]
+            else:
```

Comment:
why will this be None? can you describe the scenarios?

Reply:
when cost is not enabled and msi mode: 
![image](https://github.com/Azure/azure-cli/assets/25573844/ec586e88-18b6-4d6b-b857-c2416e99dac1)